src/unix/system.c

```diff
@@ -341,7 +344,16 @@ void Sys_Init(void)
     if (!homedir) {
 	    Sys_Error("Homedir not found!\n");
     }
-    sprintf(homegamedir, "%s/%s", homedir, ".quake2rtx");
+    xdg_data_home_dir = getenv("XDG_DATA_HOME");
+    if (xdg_data_home_dir == NULL) {
+	    xdg_data_home_dir = malloc(sizeof(char) * 200);
+	    malloc_needed = 0;
+	    sprintf(xdg_data_home_dir, "%s/%s", homedir, ".local/share");
+    }
+    check_homegamedir = sprintf(homegamedir, "%s/%s", xdg_data_home_dir, "quake2rtx");
+    if (malloc_needed && (check_homegamedir > 0)) {
+	    free(xdg_data_home_dir);
```

Comment:
Freeing xdg_data_home_dir shouldn't be bound to whether the previous sprintf() succeeded or not.